llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -1948,6 +1948,9 @@ class BoUpSLP {
 
     /// A vector of operand vectors.
     SmallVector<OperandDataVec, 4> OpsVec;
+    /// When VL[0] is IntrinsicInst, Arg_size is CallBase::arg_size. When VL[0]
+    /// is not IntrinsicInst, Arg_size is User::getNumOperands.
+    unsigned Arg_size;
```

Comment:
Why the original code cannot be reused, why need an extra data member here? Why just getNumOperands() or 2 does not work?

Reply:
For `IntrinsicInst` with more than 2 operands, `VLOperands::appendOperandsOfVL` stores the first two operands but `TreeEntry::setOperandsInOrder` stores all of the operands.
Only the first two operands should be reorder. But all of the operands can be passed into `buildTree_rec`.
If we don't use a new data member (`Arg_size`), either reorder swap non-swappable operands or we need to keep `TreeEntry::setOperandsInOrder`.

Reply:
Just mark such operands as non-reordable

Reply:
You are right. We can use 2 here.

Reply:
> Just mark such operands as non-reordable

Mark operands as non-reordable also requires additional data member to store the information.